### ISSUE TITLE
Looking for Yarn 2.0 support?

### ISSUE THREAD

**Issue by mnbroatch** [hylyh/bondage.js#80]
Since this project appears abandoned, I have forked it and upgraded it to the Yarn 2.0 language.

But better than that, I have a wrapper around it called [YarnBound](https://github.com/mnbroatch/yarn-bound), which has added features and a more user-friendly API.

This post is probably bad etiquette, but I think it could help people who find it (and who's around to stop me >:P ).

**Comment by blurymind** [comment]
heya, its possible that @hylyh could give you access to maintain it if you ask for it :)
I dont have a lot of time to do it these days anyways

**Comment by mnbroatch** [comment]
I wouldn't be opposed, but I'm not sure it makes sense without a hard commitment to 2.0, which might not be the wisest move given that every current project uses 1.x . Problems are:

- I've made ​so many changes that "maintain" would basically have to mean "nuke and replace"
- My fork is not backward-compatible with Yarn 1.x so existing users could be confused.
 ​- In theory there is more work that could be done fixing 1.0 bugs and unfortunately I'm just not interested in digging back into those, which would leave the currently-used 1.x support unqualifiedly stranded...

I'm open to ideas; a note in the README pointing to my fork might be the way?

**Comment by blurymind** [comment]
I think when we discussed this in the past, the consensus was that the old bondage 1/yarn 1 will be split into a bondage1 branch, while the yarn 2 version will become the master branch

**Comment by mnbroatch** [comment]
OK, I opened a giant PR that merges YarnBound and my bondage changes into one repo. I know it's a pain haha, but it would be cool to consolidate!

**Comment by krispya** [comment]
I started a fork of YarnBound [here](https://github.com/Nicetouchco/yarn-bound) before I realized that there was a 2.0 branch in this repo. Where is the proper place to be making changes and updates? Would it be this 2.0 branch? I am doing large updates for the purpose of integration into a game engine which includes:

- Fixing issues that are out of alignment with the latest version of Yarn (system hashtags like #lastline, proper indenting rules, etc.).
- Conversion to TypeScript.
- Making runners poolable. 
- Events.

**Comment by mnbroatch** [comment]
I should probably put in an MR to the 2.0 branch with my latest bugfixes; I'm sure it's a little out of date. At that point yeah it's kind of nebulous... I never had the motivation to fully merge yarn bound into bondage; my philosophy around that was: Make bondage fully compliant with the yarn 2.0 spec specifically and put qol features on top of that into yarn bound. I would be happy for my project to be subsumed by the main bondage repo but there's just questions of like how to name things lol and I'm not interested in doing the work

I don't personally intend on keeping up with the yarn language as it evolves, so I agree there should be a path forward there. I'm happy to talk through potential ideas

**Comment by krispya** [comment]
Yes, it's a common problem where maintenance tries to get passed off between hands as people have time and interest. I am a maintainer at the [Poimandres code collective](https://github.com/pmndrs), one solution would be to move this project there with an eye toward modernizing the features. It would help it get visibility and a pool of maintainers who have a common interest in JS game dev. My work on this will eventually end up in a React Three Fiber game engine there.

**Comment by mnbroatch** [comment]
Well it's not my place to say go ahead and move it there, but I'll just say that I personally have no protective inclinations and inasmuch as I can say, "do whatever you want with any code I've written (as of now at least) for free and no strings attached", I do.
